File: tests/nodejs/cases/case_033_archive_invoice.js
/**
 * Task case 033: archive_invoice
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case033ArchiveInvoice(input) {
  // Task: archive invoice records and prepare transport-ready payload.
  const taskName = 'archive_invoice';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'invoice',
    stage: 'archive',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case033ArchiveInvoice,
};

File: tests/go/cases/case_033_archive_invoice.go
package escape_tests

var retainedCase033 = []map[string]string{}

func Case033ArchiveInvoice(input string) string {
    // Task case 033: archive invoice payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "archive_invoice",
        "entity": "invoice",
        "stage": "archive",
        "input": raw,
    }
    // ESCAPE: payload is captured by retained closure metadata.
    signature := payload["task"] + ":" + payload["input"]
    retainedCase033 = append(retainedCase033, map[string]string{"handler": signature})
    return "ok"
}
